### PR TITLE
sqlcmd hangs for select query with null XML values

### PR DESCRIPTION
**Pull request by Suthapalli-Ramya-satya-vasavi-srija**
### Description
Previously the sqlcmd hangs for select query which has null XML value. This commit fixes by sending NbcRowToken when nullMapSize and simpleRowSize are equal. And for older clients of TDS 7.2 where NbcRowToken doesn't exist, sending PLP_NULL(0xffffffffffffffff) specially for variables of type XML.

### Issues Resolved

BABEL-4196

### Test Scenarios Covered ###
* **Use case based -** Added select queries which contain NULL xml. Inserted values through INSERT command and through prep-exec.

* **Boundary conditions -**

* **Arbitrary inputs -** Inserted NULL and some value for XML variables.

* **Negative test cases -** N/A

* **Minor version upgrade tests -** N/A

* **Major version upgrade tests -** N/A

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Deepesh125** [comment]
It is not clear in the description that we are fixing sender side for XML NULL values. So please update it accordingly

File: contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c [hunk 1]
```diff
@@ -2759,6 +2758,14 @@ TdsPrintTup(TupleTableSlot *slot, DestReceiver *self)
...
 						TdsPutInt16LE(0xffff);
 						break;
+					case TDS_TYPE_XML:
+
+						/*
+						 * In case of TDS version lower than or equal to 7.3A,
+						 * we need to send 0xffffffffffffffff (PLP_NULL)
+						 */
+						TdsPutUInt64LE(0xffffffffffffffff);
+						break;
```
Comment (Deepesh125) [inline contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c:2775]: Code where we calculate simpleRowSize needs to be updated according to this. And do we have existing test cases for this? if so please put a reference here and if not then plz add some

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c (hunk 1 above)
Comment (Suthapalli-Ramya-satya-vasavi-srija) [reply contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c:2775]: There are no existing tests for this case, adding them in `TestXML` test file

File: test/JDBC/expected/TestXML.out [hunk 2]
```diff
@@ -27,3 +27,15 @@ xml
 ~~END~~
 
 DROP TABLE XML_dt;
+
+CREATE TABLE [dbo].[notification_definition]([id] [bigint] NOT NULL,[name_tx] [nvarchar](200) NOT NULL,[description_tx] [nvarchar](max) NULL,[schedule_xml] [xml] NULL,[default_sender_tx] [nvarchar](200) NULL,[lock_id] [tinyint] NOT NULL,[settings_xml] [xml] NULL,[agency_id] [bigint] NULL)
```
Comment (Deepesh125) [inline test/JDBC/expected/TestXML.out:31]: Have a test to insert NULL through prep-exec also

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: test/JDBC/expected/TestXML.out (hunk 2 above)
Comment (Suthapalli-Ramya-satya-vasavi-srija) [reply test/JDBC/expected/TestXML.out:31]: Added tests which insert NULL through prep-exec in the next revision

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**